app/scripts/modules/core/src/cluster/allClusters.controller.js

```diff
@@ -101,18 +104,30 @@ module.exports = angular.module('spinnaker.core.cluster.allClusters.controller',
 
     this.createServerGroup = function createServerGroup() {
       providerSelectionService.selectProvider(app, 'serverGroup').then(function(selectedProvider) {
-        let provider = cloudProviderRegistry.getValue(selectedProvider, 'serverGroup');
-        $uibModal.open({
-          templateUrl: provider.cloneServerGroupTemplateUrl,
-          controller: `${provider.cloneServerGroupController} as ctrl`,
-          size: 'lg',
-          resolve: {
-            title: function() { return 'Create New Server Group'; },
-            application: function() { return app; },
-            serverGroup: function() { return null; },
-            serverGroupCommand: function() { return serverGroupCommandBuilder.buildNewServerGroupCommand(app, selectedProvider); },
-            provider: function() { return selectedProvider; }
-          }
+        versionSelectionService.selectVersion(selectedProvider).then(function(selectedVersion) {
+          let provider = cloudProviderRegistry.getValue(selectedProvider, 'serverGroup', selectedVersion);
+          $uibModal.open({
+            templateUrl: provider.cloneServerGroupTemplateUrl,
+            controller: `${provider.cloneServerGroupController} as ctrl`,
+            size: 'lg',
+            resolve: {
+              title: function () {
+                return 'Create New Server Group';
+              },
+              application: function () {
+                return app;
+              },
+              serverGroup: function () {
+                return null;
+              },
+              serverGroupCommand: function () {
+                return serverGroupCommandBuilder.buildNewServerGroupCommand(app, selectedProvider);
+              },
+              provider: function () {
+                return selectedProvider;
+              }
```

Comment:
while you're in here, it would be nice to convert these to arrow functions:
```javascript
resolve: {
  title: () => 'Create New Server Group',
  application: () => app,
  serverGroup: () => null,
  serverGroupCommand: () => serverGroupCommandBuilder.buildNewServerGroupCommand(app, selectedProvider),
  provider: () => selectedProvider
```